liquidctl/driver/asetek_pro.py

```diff
@@ -0,0 +1,256 @@
+        size and read data in chunks.  However, leviathan and its derivatives
+        seem to work fine without this complexity; we currently try the same
+        approach.
+        """
+        msg = self.device.read(_PRO_READ_ENDPOINT, length, _READ_TIMEOUT)
+        _LOGGER.debug('received %s', LazyHexRepr(msg))
+        self.device.release()
+        return msg
+
+    def _write_color_change_speed(self, mode, speed):
+        """Send the speed to cycle colors on the RGB pump"""
+        if speed < 1 or speed > 3:
+            raise ValueError('Unsupported speed ({}), only 1-3 supported'.format(speed))
+
+        self._write([0x53, _PRO_SPEEDS[mode][int(speed)-1]])
+        self._end_transaction_and_read(3)
+
+    def _write_color(self, colors):
+        """Write a list of colors to cycle between, the cycle pattern comes from the
+        value used for cycle speed"""
+        colorCount = 0
+        setColors = []
+        for color in colors:
+            colorCount = colorCount + 1
+            setColors = setColors + color
+        self._write([_PRO_CMD_WRITE_COLOR_LIST, colorCount] + setColors)
+        self._end_transaction_and_read(3)
+
+    def _get_fan_speeds(self):
+        """Read the RPM speed of the fans. Try to get the speeds for 3 fans as that is
+        how many the H150i PRO supports"""
+        speeds = []
+        for i in [0, 1, 2]:
+            self._write([_PRO_CMD_READ_FAN_SPEED, i])
+            msg = self._end_transaction_and_read(6)
+            if msg[0] != 0x41 or msg[1] != 0x12 or msg[2] != 0x34 or msg[3] != i:
+                _LOGGER.debug('Unable to get speed for fan id %d', i)
+                continue
+            speeds.append((i, (msg[4] << 8) + msg[5]))
+        return speeds
+
+    def _get_fan_indexes(self, channel):
+        if len(channel) > 3:
+            return [int(channel[3:]) - 1]
+        return [reading[0] for reading in self._get_fan_speeds()]
+
+    def initialize(self, pump_mode=None, **kwargs):
+        """Initialize the device."""
+        _LOGGER.debug('Pro configure device...')
+        if pump_mode == None:
+            return
+        else:
+            _LOGGER.debug('pump_mode %s', pump_mode)
+            self._write([_PRO_CMD_WRITE_PUMP_MODE, _PRO_PUMP_MODES.index(pump_mode)])
+            self._end_transaction_and_read(5)
+
+    def get_status(self, **kwargs):
+        """Get a status report.
+        Returns a list of `(property, value, unit)` tuples.
+        """
+        status = []
+        ### Can some Pro devices have more than one temp sensor?   OCL seems to code for that?
+        self._write([_PRO_CMD_READ_AIO_TEMP])
+        msg = self._end_transaction_and_read(6)
+        aio_temp = msg[3] + msg[4]/10
+        status.append(('Liquid temperature', aio_temp, '°C'))
+        speeds = self._get_fan_speeds()
+        for reading in speeds:
+            status.append(('Fan {} speed'.format(reading[0]), reading[1] | 0, 'rpm'))
```

Comment:
I see you're expecting `reading[1]` to be `None` sometimes.  Why would that happen?

By the way, you can use `f` strings now.  It's possible that the driver you based this on is from when liquidctl supported Python < 3.6.

Reply:
It should never be `None`, I've removed the ` | 0`. I've also moved to `f` strings